platform/android/file_access_filesystem_jandroid.cpp

```diff
@@ -72,7 +72,14 @@ Error FileAccessFilesystemJAndroid::_open(const String &p_path, int p_mode_flags
 		env->DeleteLocalRef(js);
 
 		if (res <= 0) {
-			return ERR_FILE_CANT_OPEN;
+			switch (res) {
+				case 0:
+				default:
+					return ERR_FILE_CANT_OPEN;
+
+				case -1:
+					return ERR_FILE_NOT_FOUND;
+			}
 		}
```

Comment:
Nitpick but a two-case switch inside an if for the same value seems a bit convoluted, it could just be:
```
if (res == -1) {
	return ERR_FILE_NOT_FOUND;
} else if (res <= 0) {
	return ERR_FILE_CANT_OPEN;
}
```
(assuming that it can actually be < -1)

But that's mostly a matter of taste I guess, it's fine to keep as is if you prefer it so.

Reply:
Reading the Kotlin code I'm assuming you're doing it this way as you might add further negative error codes in the future, in which case it makes sense.

Reply:
I want with something similar to what you suggested at first, but ended up preferring the `switch` statement as it makes (imo) the case handling clearer, especially if we add support for other error messages.